### DISCUSSION TITLE
Allow `--period` filtering by week and quarter

### DISCUSSION THREAD

**Discussion by chairmank** [jotaen/klog#361] [Feature ideas & discussions]
Using the `--period=PERIOD` filter, the user can select records in one year or one month. I would like to add the ability to select records in one week. A week period can be decoded from a ISO 8601 week date representation:

```
klog report --period=2022-W14 @alias
```

This enhancement would fill a gap in functionality between the shortcut `--this-week` and `--last-week` filters and the more general `--period` period.

I see a potential interaction with https://github.com/jotaen/klog/issues/141.

**Comment by jotaen** [comment]
👍 Additionally, it should also support quarters (`2022-Q1`). Even though the quarter notation is not part of ISO, it would be both useful and in line with the other filters.
